cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -83,7 +84,11 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger) (*grpc.ClientConn, er
 	}
 
 	if b.notifier != nil {
-		return nil, errors.New("not implemented")
+		logger.Info("Using external discovery service with roundrobin load balancer")
+		grpcResolver := grpcresolver.New(b.notifier, logger, b.subsetSize)
+		resolver.Register(grpcResolver)
```

Comment:
is this setting something global? Can we do it without affecting global state?

Reply:
it's actually already set within `grpcresolver.New`, I will remove it here in followup commit.